File: ОП/Lab6/Lab6 C++/Laboratorna6/laboratorna6.cpp
#include <iostream>
#include <cmath>
using namespace std;

int n;         //n - global variable

double sqrt(double a, double k);

int main() {
	double a;
	cout << "Enter (a>0) a=";
	cin >> a;
	cout << "Enter n=";
	cin >> n;

	cout << "Result is: " << (sqrt(a, 3) - sqrt(a * a + 1, 6)) / (1 + sqrt(3 + a, 7));
	
}

double sqrt(double a, double k) {
	int i;
	double y_n = a;                 //y0
	
	for (i = 1; i <= n; i = i+1) {
		y_n = (1 / k) * (a / (pow(y_n, k - 1)) + (k - 1) * y_n);
		
	}
	return y_n;              
}

File: ОП/Lab6/Lab6 Python/Laboratorna6.py
def sqrt(a,k,n):
    y_n=a
    for i in range(1,n):
        y_n=(1/k)*(a/(y_n**(k-1))+(k-1)*y_n)
        i=i+1
    return y_n
a=float(input("Enter (a>0) a="))
n=int(input("Enter n="))
print("Result is:"+str((sqrt(a,3,n)-sqrt(a*a+1,6,n))/(1+sqrt(3+a,7,n))))